pkg/patterns/addon/application.go

```diff
@@ -42,8 +43,25 @@ const (
 
 // TransformApplicationFromStatus modifies the Application in the deployment based off the CommonStatus
 func TransformApplicationFromStatus(ctx context.Context, instance declarative.DeclarativeObject, objects *manifest.Objects) error {
-	addonObject, ok := instance.(addonsv1alpha1.CommonObject)
-	if !ok {
+	var version string
+	var healthy bool
+	var addonObject addonsv1alpha1.CommonObject
+
+	if unstruct, ok := instance.(*unstructured.Unstructured); ok {
```

Comment:
Looking at the other PRs, we might want to create helper functions here.  e.g. `GetVersion` and `GetHealthy` (or maybe `GetCommonSpec` and `GetCommonStatus`)

Reply:
I could definitely work on this. I think it is a good idea